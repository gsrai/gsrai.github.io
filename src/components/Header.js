import React from 'react'
import { Link } from 'react-router-dom'

export default class Header extends React.Component {
  getPath = () => window.location.hash.slice(2)

  render() {
    const activeTab = this.getPath()
    return (
      <div>
        <header>
          <h1>Gagondeep Srai</h1>
          <nav>
            <ul>
              <li className={activeTab === '' ? 'active' : ''}>
                <Link to='/'>Projects</Link>
              </li>
              <li className={activeTab === 'about' ? 'active' : ''}>
                <Link to='about'>About</Link>
              </li>
              <li className={activeTab === 'blog' ? 'active' : ''}>
                <Link to='blog'>Blog</Link>
              </li>
            </ul>
          </nav>
        </header>
      </div>
    )
  }
}
