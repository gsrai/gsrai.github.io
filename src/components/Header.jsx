import React from 'react';
import PropTypes from 'prop-types';

export default class Header extends React.Component {
  constructor(props) {
    super(props);
  }

  onProjectClick = (e) => {
    e.preventDefault();
    this.props.onNavChange('projects');
  }

  onAboutClick = (e) => {
    e.preventDefault();
    this.props.onNavChange('about');
  }

  onBlogClick = (e) => {
    e.preventDefault();
    this.props.onNavChange('blog');
  }

  render() {
    const activeTab = this.props.tabName;
    return (
      <div>
        <header>
          <h1>Gagondeep Srai</h1>
          <nav>
            <ul>
              <li className={activeTab === 'projects' ? 'active': ''}>
                <a href="#" onClick={this.onProjectClick} >Projects</a>
              </li>
              <li className={activeTab === 'about' ? 'active': ''}>
                <a href="#" onClick={this.onAboutClick}>About</a>
              </li>
              <li className={activeTab === 'blog' ? 'active': ''}>
                <a href="#" onClick={this.onBlogClick}>Blog</a>
              </li>
            </ul>
          </nav>
        </header>
      </div>
    );
  }

}

Header.propTypes = {
  tabName: PropTypes.string.isRequired,
  onNavChange: PropTypes.func.isRequired
};