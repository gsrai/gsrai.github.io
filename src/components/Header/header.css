.center-nav {
  top: 0;
  position: fixed;
}

header > h2 {
  margin: 0;
}