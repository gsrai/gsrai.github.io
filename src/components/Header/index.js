import React from 'react'
import { Link } from 'react-router-dom'

import './header.css'

export default class Header extends React.Component {
  getPath = () => window.location.hash.slice(2)

  isActive = (isActive) => {
    const normalClassname = 'paper-btn margin'
    return isActive ? normalClassname + ' btn-success' : normalClassname
  }

  render() {
    const activeTab = this.getPath()
    return (
      <React.Fragment>
        <header className='row flex-center'>
          <h2>Gagondeep Srai</h2>
        </header>
        <div className='row flex-center'>
          <Link className={this.isActive(activeTab === '')} to='/'>Projects</Link>
          <Link className={this.isActive(activeTab === 'about')} to='/about'>About</Link>
          <Link className={this.isActive(activeTab === 'blog')} to='/blog'>Blog</Link>
        </div>
      </React.Fragment>
    )
  }
}
