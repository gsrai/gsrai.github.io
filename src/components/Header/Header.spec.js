import { shallow, mount } from 'enzyme'
import React from 'react'
import { MemoryRouter, Link } from 'react-router-dom'
import Header from './index'

describe('Component Rendering', () => {
  test('component should render correctly', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
  })

  test('renders Links correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    )

    const linksTo = wrapper.find(Link).map(l => l.props().to)
    expect(linksTo).toEqual(['/', '/about', '/blog'])
  })
})

describe('Component Behavior', () => {
  test('Active link should have correct className', () => {
    const classname = '.btn-success'
    const wrapper = mount(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    )

    expect(wrapper.find(classname).exists()).toBe(true)
  })
})
