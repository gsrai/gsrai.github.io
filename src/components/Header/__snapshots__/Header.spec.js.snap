// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Rendering component should render correctly 1`] = `
<Fragment>
  <header
    className="row flex-center"
  >
    <h2>
      Gagondeep Srai
    </h2>
  </header>
  <div
    className="row flex-center"
  >
    <Link
      className="paper-btn margin btn-success"
      replace={false}
      to="/"
    >
      Projects
    </Link>
    <Link
      className="paper-btn margin"
      replace={false}
      to="/about"
    >
      About
    </Link>
    <Link
      className="paper-btn margin"
      replace={false}
      to="/blog"
    >
      Blog
    </Link>
  </div>
</Fragment>
`;
